build.rs

```diff
@@ -122,6 +122,12 @@ fn main() -> Result<(), Box<dyn std::error::Error>> {
         "{ let a = Some(5i32); matches!(a, None) }",
     );
 
+    emit_expression_maybe_using_feature(
```

Comment:
It might be worth it to add a comment explaining since when this is supported so we can take it into account when discussing the minimum rust version required.

Reply:
MSRV is there to "be a good ecosystem citizen" and consider compatibility at a quick glance, and when modified it is an important change, but I prefer to have people "find out" a new MSRV as they remove tests in this file rather than "knowing" that since they removed a given feature then the MSRV "must be" some specific version attached to the test in a comment (which is dangerous as implementations of Rust like `gcc-rs` would likely not follow the same evolution). Note that even compilers with a version lower than the MSRV can compile the library (ie. nightlies and betas), and we don't force a minimum version (we just warn that compilation might fail to let the use have a hint at the cause of a potential breakage).

The whole idea behind this mechanism is that you don't need nor want to check the versions at all - things are tested for you. I prefer not to pin any such comments to features because it's easy for people to start thinking about versions and not about capabilities, and start trusting the comments instead of actually running the tests.